src/liballoc/rc.rs

```diff
@@ -571,9 +571,14 @@ impl<T: ?Sized> Rc<T> {
     /// ```
     #[stable(feature = "rc_raw", since = "1.17.0")]
     pub fn into_raw(this: Self) -> *const T {
-        let ptr: *const T = &*this;
+        let ptr: *mut RcBox<T> = NonNull::as_ptr(this.ptr);
+        let fake_ptr = ptr as *mut T;
         mem::forget(this);
```

Comment:
`mem::forget(this);` takes ownership of struct containing a pointer that `ptr` has already taken ownership of. For this pattern, `ManuallyDrop` is to be preferred:

```rust
let ptr: *mut RcBox<T> = NonNull::as_ptr(ManuallyDrop::new(this).ptr);
let fake_ptr ...
```

Reply:
I'm just using the same way of forgetting the destructor as already used in these files. (FWIW, I plan to take another pass that adds `as_raw` impls and uses `ManuallyDrop` to clean it up a little bit.)

Reply:
Well but if you are touching this code anyway why not use `ManuallyDrop` while you are at it?

Reply:
Mainly because I plan to touch more of this file in a bigger refactor in a follow up PR and I don't want to clutter this PR and make it harder to r+.

If I need to adjust the patch again I'll go ahead and use manuallydrop in the touched functions.